src/CardPage.js

```diff
@@ -0,0 +1,268 @@
+                <div className='mdc-card__media mdc-card__media--square demo-card__media demo-card__media--music'
+                     style={{backgroundImage: `url("${cardMedia1x1Img}")`}}></div>
+                <div className='demo-card__music-info'>
+                  <div className='demo-card__music-title mdc-typography--headline5'>Rozes</div>
+                  <div className='demo-card__music-artist mdc-typography--body2'>Under the Grave</div>
+                  <div className='demo-card__music-year mdc-typography--body2'>(2016)</div>
+                </div>
+              </div>
+            </div>
+            <hr className='mdc-list-divider'/>
+            <div className='mdc-card__actions'>
+              <div className='mdc-card__action-buttons demo-card__action-buttons--text-only'>Rate this album</div>
+              <div className='mdc-card__action-icons'>
+                <i className='material-icons demo-card__action-icon--star mdc-ripple-surface'
+                   tabIndex='0'
+                   role='button'
+                   title='1 star'
+                   data-mdc-ripple-is-unbounded
+                   ref={this.initRipple}>
+                  star_border
+                </i>
+                <i className='material-icons demo-card__action-icon--star mdc-ripple-surface'
+                   tabIndex='0'
+                   role='button'
+                   title='2 stars'
+                   data-mdc-ripple-is-unbounded
+                   ref={this.initRipple}>
+                  star_border
+                </i>
+                <i className='material-icons demo-card__action-icon--star mdc-ripple-surface'
+                   tabIndex='0'
+                   role='button'
+                   title='3 stars'
+                   data-mdc-ripple-is-unbounded
+                   ref={this.initRipple}>
+                  star_border
+                </i>
+                <i className='material-icons demo-card__action-icon--star mdc-ripple-surface'
+                   tabIndex='0'
+                   role='button'
+                   title='4 stars'
+                   data-mdc-ripple-is-unbounded
+                   ref={this.initRipple}>
+                  star_border
+                </i>
+                <i className='material-icons demo-card__action-icon--star mdc-ripple-surface'
+                   tabIndex='0'
+                   role='button'
+                   title='5 stars'
+                   data-mdc-ripple-is-unbounded
+                   ref={this.initRipple}>
+                  star_border
+                </i>
+              </div>
+            </div>
+          </div>
+
+          <div className='mdc-card mdc-card--outlined demo-card'>
+            <div className='demo-card-article-group-heading mdc-typography--subtitle2'>Headlines</div>
+
+            <hr className='mdc-list-divider'/>
+
+            <div className='demo-card-article mdc-ripple-surface' tabIndex='0' ref={this.initRipple}>
```

Comment:
Here are 3 of the same node structures. You could make another component here and no need to have a constructor in the parent component.

```
class CardArticle extends Component {
  ripple = null;
  
  componentWillUnmount() {
    this.ripple.destroy();
  }
  render() {
    return (
      <div
        className='demo-card-article mdc-ripple-surface'
        tabIndex='0' 
        ref={(surfaceEl) => this.ripple = new MDCRipple(surfaceEl)}
      >
        <h2 className='demo-card-article__title mdc-typography--headline5'>{this.props.title}</h2>
        <p className='demo-card-article__snippet mdc-typography--body2'>
          {this.props.subtitle}
        </p>
      </div>
    );
  }
}
```